### PR TITLE
use vformat for duplicate strings

### LINKED ISSUE #30111: Needlessly repeated texts in VisualShader translations
**Godot version:**
3.1.1

**Issue description:**
While translating on Weblate, I encountered a streak of same-looking texts, which differed only in an untranslatable string. The texts are e.g.:
`'uv' input parameter for all shader modes.`

My point is, in above text the "uv" will probably stay the same in all languages, as I assume it's a constant name. There are 10 texts like this and they could be simply:
`'%s' input parameter for all shader modes.`

Later there's also
`'%s' input parameter for vertex and fragment shader modes.`
`'%s' input parameter for fragment and light shader modes.`
and few more similar groups. Using a format instead would result in much less texts to translate, because it's pointless to keep them as separate strings.

### PR DESCRIPTION
**Pull request by hbina**
many strings in visual_shader_editor_plugin.cpp does not use vformat for duplicate strings making it harder to translate

Fixes #30111

### CODE REVIEW

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -2050,104 +2050,104 @@ VisualShaderEditor::VisualShaderEditor() {
...
-	add_options.push_back(AddOption("Projection", "Input", "All", "VisualShaderNodeInput", TTR("'projection' input parameter for all shader modes."), "projection", VisualShaderNode::PORT_TYPE_TRANSFORM, -1, Shader::MODE_SPATIAL));
-	add_options.push_back(AddOption("Time", "Input", "All", "VisualShaderNodeInput", TTR("'time' input parameter for all shader modes."), "time", VisualShaderNode::PORT_TYPE_SCALAR, -1, Shader::MODE_SPATIAL));
-	add_options.push_back(AddOption("ViewportSize", "Input", "All", "VisualShaderNodeInput", TTR("'viewport_size' input parameter for all shader modes."), "viewport_size", VisualShaderNode::PORT_TYPE_VECTOR, -1, Shader::MODE_SPATIAL));
-	add_options.push_back(AddOption("World", "Input", "All", "VisualShaderNodeInput", TTR("'world' input parameter for all shader modes."), "world", VisualShaderNode::PORT_TYPE_TRANSFORM, -1, Shader::MODE_SPATIAL));
+	add_options.push_back(AddOption("Camera", "Input", "All", "VisualShaderNodeInput", vformat(TTR("'%s' input parameter for all shader modes."), "camera"), "camera", VisualShaderNode::PORT_TYPE_TRANSFORM, -1, Shader::MODE_SPATIAL));
```
Comment (akien-mga) [inline editor/plugins/visual_shader_editor_plugin.cpp]: Maybe the `TTR()` call could be factored out in a const String before its first use, and then do `vformat(input_param_desc, "camera")`, etc. ?

Comment (hbina) [reply editor/plugins/visual_shader_editor_plugin.cpp]: yep, that would be smarter. working on it. Does the code style use lowercase for what I consider to be a constant?

Comment (akien-mga) [reply editor/plugins/visual_shader_editor_plugin.cpp]: Well we use uppercase identifiers for global/class constants, but here it should just be a llvariable in the local scope that happens to be const, so I don't think it should be uppercase.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**